frontend/src/components/TitleBar/Canvas/CanvasCalendar.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { ReactElement, Component } from 'react';
+import settingStyles from '../Settings/SettingsPage.module.css';
+import styles from './CanvasCalendar.module.css';
+import { setCanvasCalendar } from '../../../firebase/actions';
+
+type Props = { readonly linked: boolean }
+
+type State = { canvasCalendar: string | null }
+
+export default class CanvasCalendar extends Component<Props, State> {
+  public state: State = { canvasCalendar: '' };
```

Comment:
Why don't we simply pass `canvasCalendar` down and don't keep any state in this component?

Reply:
Yep, not having a state would be best. The iCal link that the user inputs is stored in `canvasCalendar` which is passed to `setCanvasCalendar()`. I'm not sure how to get the current value in a form field without storing it in a state.

Reply:
We seems to be hit by the same `TaskEditor` problem here. Duplicating the link might be necessary.
However, currently, the user can't see the link unless they unlink first.

Reply:
Then should I display the iCal link below the "Your Canvas calendar feed is linked" text?

Reply:
Looks good to me